clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -584,27 +584,6 @@ public static String formatBytes(long bytes) {
         }
     }
 
-    /**
-     * Create a string representation of an array joined by the given separator
-     * @param strs The array of items
-     * @param separator The separator
-     * @return The string representation.
-     */
-    public static <T> String join(T[] strs, String separator) {
-        return join(Arrays.asList(strs), separator);
-    }
-
-    /**
-     * Create a string representation of a collection joined by the given separator
-     * @param collection The list of items
-     * @param separator The separator
-     * @return The string representation.
-     */
-    public static <T> String join(Collection<T> collection, String separator) {
-        Objects.requireNonNull(collection);
-        return mkString(collection.stream(), "", "", separator);
-    }
-
     /**
      * Create a string representation of a stream surrounded by `begin` and `end` and joined by `separator`.
      *
```

Comment:
It seems `mkString(Stream<T> stream, String begin, String end, String separator)` can be removed too since the only one use case (https://github.com/apache/kafka/blob/5de5d967adffd864bad3ec729760a430253abf38/tools/src/test/java/org/apache/kafka/tools/consumer/group/ConsumerGroupServiceTest.java#L199) does not use "begin" and "end"

Reply:
There are only 11 use cases in code base, so we can do the cleanup in this PR.

Reply:
Thanks for the reply. Suggestions applied.